clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -9159,7 +9159,27 @@ void OffloadBundler::ConstructJob(Compilation &C, const JobAction &JA,
                    ? Action::GetOffloadKindName(Action::OFK_SYCL)
                    : Action::GetOffloadKindName(CurKind);
     Triples += '-';
-    Triples += CurTC->getTriple().normalize();
+    // Incoming DeviceArch is set, break down the Current triple and add the
+    // device arch value to it.
+    // This is done for AOT targets only.
+    std::string DeviceArch;
+    llvm::Triple TargetTriple(CurTC->getTriple());
+    bool IsAOT = TargetTriple.getSubArch() == llvm::Triple::SPIRSubArch_fpga ||
```

Comment:
Could we make the isAOT computation a utility function? I see it used in a few places.

Reply:
Yeah, I can make adjustments here - there are uses of `IsAOT` and `IsSpirvAOT` that can be cleaned up.